v2/AdminCore.Services/AuthenticatedUser.cs

```diff
@@ -0,0 +1,34 @@
+﻿using AdminCore.Common.Interfaces;
+using Microsoft.AspNetCore.Http;
+using System.Security.Claims;
+using AdminCore.Common;
+using AdminCore.DTOs.Employee;
+
+namespace AdminCore.Services
+{
+  public class AuthenticatedUser : IAuthenticatedUser
+  {
+    private readonly IHttpContextAccessor _httpContentAccessor;
+    private readonly IEmployeeService _employeeService;
+
+    public AuthenticatedUser(IHttpContextAccessor httpContextAccessor, IEmployeeService employeeService)
+    {
+      _httpContentAccessor = httpContextAccessor;
+      _employeeService = employeeService;
+    }
+
+    public EmployeeDto RetrieveLoggedInUser()
+    {
+      var userDetails = GetLoggedInUserDetails();
```

Comment:
Might be worth doing 2 things here:
1. Checking does that key exist in the dictionary.
2. store the key itself in a constants file (along with any other details we may query from the dictionary).